docs/release-notes/release-notes.md

```diff
@@ -22,6 +22,60 @@ The `rpc.grpc_laddr` config option is now required when running the celestia-app
 grpc_laddr = "tcp://127.0.0.1:9098"
 ```
 
+### State Machine Changes (v4.0.0)
+
+Celestia-app v4.0.0 includes significant state machine changes due to major dependency upgrades: **Cosmos SDK** (v0.46.16 → v0.50.13), **IBC** (v6.2.2 → v8.7.0), and **CometBFT** (v0.34 → v0.38).
+
+#### New Messages (Added Modules)
+
+**`x/circuit` Circuit Breaker Module** ([cosmos-sdk docs](https://docs.cosmos.network/v0.50/build/modules/circuit)):
+- `MsgAuthorizeCircuitBreaker` - Grant circuit breaker permissions
+- `MsgTripCircuitBreaker` - Disable message execution 
+- `MsgResetCircuitBreaker` - Re-enable message execution
+
+**`x/consensus` Consensus Parameters Module** ([cosmos-sdk docs](https://docs.cosmos.network/v0.50/build/modules/consensus)):
+- `MsgUpdateParams` - Update consensus parameters via governance (replaces CometBFT consensus param updates)
+
+**Hyperlane Interoperability**:
+- `hyperlane/core` - Cross-chain messaging infrastructure
+- `hyperlane/warp` - Token bridging and routing
+
+#### Removed Messages (Deprecated Modules)
+
+**`x/crisis` Module** - Removed in Cosmos SDK v0.50.x ([migration docs](https://docs.cosmos.network/v0.50/build/migrations/upgrading#xconsensus)):
+- `MsgVerifyInvariant` - Emergency chain halt functionality moved to circuit breaker
```

Comment:
Seems incorrect. The circuit breaker can only disable messages not halt the chain
```suggestion
```